### PR TITLE
d_a_ikari OK

### PR DESCRIPTION
**Pull request by LagoLunatic**

### CODE REVIEW

File: include/d/actor/d_a_ikari.h [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
 #include "SSystem/SComponent/c_phase.h"
+#include "f_op/f_op_actor.h"
+#include "m_Do/m_Do_hostIO.h"
+
+class daObjIkariHIO_c : public mDoHIO_entry_c {
+public:
+    daObjIkariHIO_c() {
+        unk[0] = 0;
+        unk[1] = 0;
+        mWindPowerScale = 0.1;
```
Comment (LagoLunatic) [inline include/d/actor/d_a_ikari.h:13]:
```suggestion
        mWindPowerScale = 0.1f;
```

File: src/d/actor/d_a_ikari.cpp [hunk 2]
```diff
@@ -4,70 +4,197 @@
...
 }
 
 /* 000001C4-00000290       .text _createHeap__9daIkari_cFv */
-void daIkari_c::_createHeap() {
-    /* Nonmatching */
+bool daIkari_c::_createHeap() {
+    static int ikari_bdl[5] = {0x03, 0x03, 0x03, 0x04, 0x05};
+
+    J3DModelData* modelData = static_cast<J3DModelData*>(
+        dComIfG_getObjectRes(daIkari_c::M_arcname, ikari_bdl[mModelType]));
```
Comment (LagoLunatic) [inline src/d/actor/d_a_ikari.cpp:39]:
```suggestion
        dComIfG_getObjectRes(M_arcname, ikari_bdl[mModelType]));
```

File: src/d/actor/d_a_ikari.cpp [hunk 3]
```diff
@@ -4,70 +4,197 @@
...
+    g_env_light.setLightTevColorType(mpModel, &mTevStr);
+    mDoExt_modelUpdateDL(mpModel);
+    return true;
 }
 
 /* 00000494-000005B0       .text _create__9daIkari_cFv */
-s32 daIkari_c::_create() {
-    /* Nonmatching */
+int daIkari_c::_create() {
+    int phase = dComIfG_resLoad(&mPhs, daIkari_c::M_arcname);
```
Comment (LagoLunatic) [inline src/d/actor/d_a_ikari.cpp:113]:
```suggestion
    int phase = dComIfG_resLoad(&mPhs, M_arcname);
```

File: src/d/actor/d_a_ikari.cpp [hunk 4]
```diff
@@ -4,70 +4,197 @@
...
+
+    if (mModelType == 0xff) {
+        mModelType = 0;
+    }
+
+    if (mEnvType == 0xff) {
+        mScale.x = 1.0;
+        mScale.y = 1.0;
+        mScale.z = 1.0;
+        return;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_ikari.cpp:64]: return not necessary here

File: src/d/actor/d_a_ikari.cpp [hunk 5]
```diff
@@ -4,70 +4,197 @@
...
+bool daIkari_c::_delete() {
+    dComIfG_resDelete(&mPhs, M_arcname);
+    return true;
 }
 
 /* 000005E0-00000600       .text daIkariCreate__FPv */
-static s32 daIkariCreate(void*) {
+static s32 daIkariCreate(void* i_this) {
+    return ((daIkari_c*)i_this)->_create();
     /* Nonmatching */
```
Comment (LagoLunatic) [inline src/d/actor/d_a_ikari.cpp:154]: you left a few nonmatching commenting like this down here

File: src/d/actor/d_a_ikari.cpp [hunk 6]
```diff
@@ -4,70 +4,197 @@
...
 }
 
 /* 00000600-00000624       .text daIkariDelete__FPv */
-static BOOL daIkariDelete(void*) {
+static BOOL daIkariDelete(void* i_this) {
+    return ((daIkari_c*)i_this)->_delete() & 0xff;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_ikari.cpp:159]:
```suggestion
    return ((daIkari_c*)i_this)->_delete();
```

File: src/d/actor/d_a_ikari.cpp [hunk 7]
```diff
@@ -4,70 +4,197 @@
...
     /* Nonmatching */
 }
 
 /* 00000624-00000648       .text daIkariExecute__FPv */
-static BOOL daIkariExecute(void*) {
+static BOOL daIkariExecute(void* i_this) {
+    return ((daIkari_c*)i_this)->_execute() & 0xff;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_ikari.cpp:165]:
```suggestion
    return ((daIkari_c*)i_this)->_execute();
```

File: src/d/actor/d_a_ikari.cpp [hunk 8]
```diff
@@ -4,70 +4,197 @@
...
     /* Nonmatching */
 }
 
 /* 00000648-0000066C       .text daIkariDraw__FPv */
-static BOOL daIkariDraw(void*) {
-    /* Nonmatching */
+static BOOL daIkariDraw(void* i_this) {
+    return ((daIkari_c*)i_this)->_draw() & 0xff;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_ikari.cpp:171]:
```suggestion
    return ((daIkari_c*)i_this)->_draw();
```

File: src/d/actor/d_a_ikari.cpp [hunk 9]
```diff
@@ -4,70 +4,197 @@
...
+
+    if (mEnvType == 0xff) {
+        mScale.x = 1.0;
+        mScale.y = 1.0;
+        mScale.z = 1.0;
+        return;
+    } else if (mEnvType == 0x01) {
+        mScale.x = 1.27;
+        mScale.y = 1.27;
+        mScale.z = 1.27;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_ikari.cpp:68]: the float literals in this function should have the `f` suffix

File: src/d/actor/d_a_ikari.cpp [hunk 10]
```diff
@@ -4,70 +4,197 @@
...
+
+            fopAcM_SetMtx(this, &mpModel->mBaseTransformMtx[0]);
+            fopAcM_setCullSizeBox(this,
+                                  /* minX */ -160.0f * mScaleX,
+                                  /* minY */ -2500.0f * mScaleX,
+                                  /* minZ */ -600.0f * mScaleX,
+                                  /* maxX */ 160.0f * mScaleX,
+                                  /* maxY */ 100.0f * mScaleX,
+                                  /* maxZ */ 600.0f * mScaleX);
+            fopAcM_setCullSizeFar(this, 10.0);
```
Comment (LagoLunatic) [inline src/d/actor/d_a_ikari.cpp:136]:
```suggestion
            fopAcM_setCullSizeFar(this, 10.0f);
```

File: src/d/actor/d_a_ikari.cpp [hunk 11]
```diff
@@ -4,70 +4,197 @@
...
+
+            mTimer = (short)(int)cM_rndF(32768.0);
```
Comment (LagoLunatic) [inline src/d/actor/d_a_ikari.cpp:138]:
```suggestion
            mTimer = (short)(int)cM_rndF(32768.0f);
```

**Comment by kageurufu** [comment]
That last commit. I don't love it, but we match 🤷🏻

**Comment by LagoLunatic** [comment]
If you make `_createHeap` return `BOOL` instead of `bool`, shouldn't `createHeap_CB` then be able to have a proper return?

If that doesn't work, add a comment explaining the current code is a fakematch and what problem it's fixing

**Pull request merged**